pkg/operator/configobserver/config_observer_controller.go

```diff
@@ -96,13 +98,18 @@ func NewNestedConfigObserver(
 	listers Listers,
 	informers []factory.Informer,
 	nestedConfigPath []string,
+	conditionPrefix string,
```

Comment:
Just thinking if it would be possible to take it from `nestedConfigPath` somehow?

Reply:
for example, CAO uses `NewNestedConfigObserver(..., []string{"oauthAPIServer"}`.

So we could take the first element and make it pretty so that it ends up being  `OauthAPIServerConfigObservationDegraded`, not practical?

Reply:
I was wondering about the same, and thought whether anyone would go as deep as layer 2 different nesting. Probably not, so we can just as well use the first element, either seems fine.

Reply:
yeah, taking the first element seems to be enough.

How we will make it pretty? Is there an easy way to convert it to a camel case string?

Reply:
I don't think there's an easy way, the simplest thing we can do is just to take the first rune of the string and turn it into upper case :shrug:

Reply:
alright, so let's make it explicit.